File: leetcode/2073-time-needed-to-buy-tickets.py
# https://leetcode.com/problems/time-needed-to-buy-tickets/

from typing import List

class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        required_time = 0
        for i in range(len(tickets)):
            required_time += min(tickets[i], (tickets[k] - 1 if i > k else tickets[k]))
        return required_time

File: leetcode/2073-time-needed-to-buy-tickets.js
// https://leetcode.com/problems/time-needed-to-buy-tickets/

/**
 * @param {number[]} tickets
 * @param {number} k
 * @return {number}
 */
var timeRequiredToBuy = function(tickets, k) {
  let time = 0;
  for (let i = 0; i < tickets.length; i += 1) {
    time += Math.min(tickets[i], (i > k ? tickets[k] - 1 : tickets[k]));
  }
  
  return time;
};
